spec.bs

```diff
@@ -13,3 +13,540 @@ Group: patcg
+      // the HTTP request.
+      const response = await fetch('https://ads.example/get-creative', {browsingTopics: true});
+      const ad_creative = await response.json();
+      // Display the ad.
+    </pre>
+  </div>
+</section>
+
+<section>
+  <h2 id="terminology-and-types-header">Terminology and types</h2>
+  A <dfn for="browsing topics types">taxonomy</dfn> comprises a list of advertising <dfn for="browsing topics types">topic ids</dfn> integers. A [=browsing topics types/taxonomy=] is identified by a <dfn for="browsing topics types">taxonomy version</dfn> string. A [=browsing topics types/topic id=] is no smaller than 1.
+
+  The <dfn for="browsing topics types">model version</dfn> is a string that identifies the <dfn for="browsing topics types">model</dfn> used to <dfn>classify</dfn> a string into [=topic ids=]. The meaning may vary across browser vendors. The classification result [=topic ids=] should be relevant to the input string's underlying content.
+
+  The <dfn for="browsing topics types">configuration version</dfn> identifies the algorithm (other than the model part) used to calculate the topic. It should take the form of "<code>&lt;browser vendor identifier&gt;.&lt;an integer version&gt;</code>". The meaning may vary across browser vendors.
+
+  Given [=browsing topics types/configuration version=] |configurationVersion|, [=browsing topics types/taxonomy version=] |taxonomyVersion|, and [=browsing topics types/model version=] |modelVersion|, the <dfn for="browsing topics types">version</dfn> is the result of [=string/concatenating=] « |configurationVersion|, |taxonomyVersion|, |modelVersion| » using "<code>:</code>".
+
+  A <dfn for="browsing topics types">user topics state</dfn> is a struct with the following fields and default values:
+  - <dfn for="user topics state">epochs</dfn>: a list of [=epoch=]s, default to an empty list.
+  - <dfn for="user topics state">hmac key</dfn>: 128 bit number, default to 0.
+
+  An <dfn for="browsing topics types">epoch</dfn> is a struct with the following fields:
+  - <dfn for="epoch">taxonomy</dfn>: a list of integers.
+  - <dfn for="epoch">taxonomy version</dfn>: a string.
+  - <dfn for="epoch">model version</dfn>: a string.
+  - <dfn for="epoch">config version</dfn>: a string.
+  - <dfn for="epoch">top 5 topics with caller origins</dfn>: a list of [=topic with caller origins=].
+  - <dfn for="epoch">time</dfn>: a {{DOMHighResTimeStamp}} (from Unix epoch).
+
+  A <dfn for="browsing topics types">topic with caller origins</dfn> is a struct with the following fields:
+  - <dfn for="topic with caller origins">topic id</dfn>: an integer.
+  - <dfn for="topic with caller origins">caller origins</dfn>: a set of [=origins=].
+
+  A <dfn for="browsing topics types">topics history entry</dfn> is a struct with the following fields and default values:
+  - <dfn for="topics history entry">document id</dfn>: an integer, default to 0.
```

Comment:
Nit: you are inconsistent on capitalizing "id" or not, between the two definitions. I don't know if we have a great precedent for either when defining prose fields, but within a single spec you should pick one capitalization.